async_fuse/src/memfs/dir.rs

```diff
@@ -94,58 +153,77 @@ impl DirEntry {
 
         Self {
             ino,
-            name: name.into(),
+            name: name.to_owned(),
             entry_type,
         }
     }
 }
 
-impl Iterator for Dir {
-    type Item = nix::Result<DirEntry>;
+/// Iterator over the entries in a directory.
+pub struct ReadDir {
```

Comment:
Actually it's a Dir Iterator. Let's make the name DirIter.

Reply:
The name [`ReadDir`](https://doc.rust-lang.org/std/fs/struct.ReadDir.html)  is from the std library.

<https://rust-lang.github.io/api-guidelines/naming.html#methods-on-collections-that-produce-iterators-follow-iter-iter_mut-into_iter-c-iter>
<https://rust-lang.github.io/api-guidelines/naming.html#iterator-type-names-match-the-methods-that-produce-them-c-iter-ty>

Maybe `IntoIter` is better.